dataclients/kubernetes/resources.go

```diff
@@ -24,3 +24,10 @@ type secret struct {
 type secretList struct {
 	Items []*secret `json:"items"`
 }
+
+type objectReference struct {
+	Kind      string `json:"kind"`
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+	Uid       string `json:"uid"`
```

Comment:
this we don't need

Reply:
This is for unmarshalling k8s API response. 
uid can be used to detect pod rotation with the same name, e.g. from StatefulSet.